crates/qasm2/src/lib.rs

```diff
@@ -94,14 +96,22 @@ fn bytecode_from_string(
 /// without loading the entire token and parse tree into memory at once.
 #[pyfunction]
 fn bytecode_from_file(
+    py: Python<'_>,
     path: std::ffi::OsString,
     include_path: Vec<std::path::PathBuf>,
     custom_instructions: Vec<CustomInstruction>,
     custom_classical: Vec<CustomClassical>,
     strict: bool,
 ) -> PyResult<bytecode::BytecodeIterator> {
     bytecode::BytecodeIterator::new(
-        lex::TokenStream::from_path(path, strict)?,
+        lex::TokenStream::from_path(&path, strict).map_err(|err| {
+            let exc = QASM2ParseError::new_err(format!(
+                "failed to read a token stream from file '{}'",
```

Comment:
Super minor nitpick, you don't want to start this with a capital f?

```suggestion
                "Failed to read a token stream from file '{}'",
```